packages/shared/src/UI/components/WalletStatusBar/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { WalletMessages } from '@masknet/plugin-wallet'
+import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
+import { makeStyles, MaskColorVar, useStylesExtends } from '@masknet/theme'
+import { useAccount, useChainIdValid, useWallet } from '@masknet/web3-shared-evm'
+import { Box, Button, CircularProgress } from '@mui/material'
+import { useSharedI18N } from '../../../locales'
+import { WalletUI } from './WalletUI'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        display: 'flex',
+        alignItems: 'center',
+        width: '100%',
+        boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.05)',
+    },
+    button: {
+        borderRadius: 8,
+        position: 'relative',
+        textAlign: 'center',
+        margin: 0,
+    },
+
+    progress: {
+        color: MaskColorVar.twitterButtonText,
+        position: 'absolute',
+        top: theme.spacing(1),
+        left: `calc(50%-${theme.spacing(1)})`,
+    },
+}))
+interface WalletStatusBarProps extends withClasses<'button'> {
+    iconSize?: number
+    badgeSize?: number
+    actionProps?: {
```

Comment:
Flatten this prop?

Reply:
![image](https://user-images.githubusercontent.com/72531217/170827514-6177f23b-5db1-4a63-bc7f-5bd3237bee47.png)